Source/WTF/wtf/PlatformEnableCocoa.h

```diff
@@ -1105,3 +1105,11 @@
 #if !defined(ENABLE_CONTEXT_MENU_IMAGES_ON_MAC) && PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 260000
 #define ENABLE_CONTEXT_MENU_IMAGES_ON_MAC 1
 #endif
+
+#if ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 150500) \
+    || ((PLATFORM(IOS) || PLATFORM(MACCATALYST)) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 180500) \
+    || (PLATFORM(VISION) && __VISION_OS_VERSION_MIN_REQUIRED >= 20500) \
+    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED >= 110500) \
+    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MIN_REQUIRED >= 180500))
+#define ENABLE_NSURL_ERROR_FAILING_URL_STRING_KEY_DEPRECATION 1
```

Comment:
On second thought, I wonder if it's better to use `HAVE_NSURL_ERROR_FAILING_URL_STRING_KEY` or `USE_NSURL_ERROR_FAILING_URL_STRING_KEY`. What do you think?